apps/meteor/tests/unit/server/livechat/hooks/beforeNewRoom.spec.ts

```diff
@@ -0,0 +1,53 @@
+import { expect } from 'chai';
+import { describe, it } from 'mocha';
+import proxyquire from 'proxyquire';
+import sinon from 'sinon';
+
+import { callbacks } from '../../../../../lib/callbacks';
+
+proxyquire.noCallThru().load('../../../../../ee/app/livechat-enterprise/server/hooks/beforeNewRoom.ts', {
+	'meteor/meteor': {
+		Meteor: {
+			Error,
+		},
+	},
+	'@rocket.chat/models': {
+		OmnichannelServiceLevelAgreements: {
+			findOneByIdOrName: sinon.stub().callsFake((id) => {
+				if (id === 'high') {
+					return Promise.resolve({ _id: 'high' });
+				}
+				return Promise.resolve(null);
+			}),
```

Comment:
A suggestion, you can create the `sinon.stub()` outside of here, and then on tests do something like `myStub.withArgs('high').resolves({_id: 'high'})`

should look cleaner

Reply:
If you go this way, don't forget to `reset` your stub before using them again on other tests (to avoid conflicts)